Bottom: f494aa70fad442a74b4b42e4cc58eabbc4dd57e4
Top:    8473a766eb884ac93de75b3ee9b51d0c881449c1
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-10-11 22:31:45 +0100

Add support for mice / touchpads with a left/right scroll-wheel

Tested-by: Dave McGuire <mcguire@neurotica.com>


---

diff --git a/src/gpcb-menu.res.in b/src/gpcb-menu.res.in
index 7e63811..7dee9b9 100644
--- a/src/gpcb-menu.res.in
+++ b/src/gpcb-menu.res.in
@@ -37,6 +37,13 @@ Mouse =
 # If you want zoom to center, do this instead.
   #Up = { {Zoom(0.8) Center()} }
   #Down = { {Zoom(1.25) Center()} }
+
+  scroll-left = {
+    Scroll(left)
+  }
+  scroll-right = {
+    Scroll(right)
+  }
 }
 
 MainMenu =
diff --git a/src/hid/common/hid_resource.c b/src/hid/common/hid_resource.c
index 842a058..a98dc46 100644
--- a/src/hid/common/hid_resource.c
+++ b/src/hid/common/hid_resource.c
@@ -71,6 +71,10 @@ button_name_to_num (const char *name)
     return 4;
   else if (strcasecmp (name, "down") == 0)
     return 5;
+  else if (strcasecmp (name, "scroll-left") == 0)
+    return 6;
+  else if (strcasecmp (name, "scroll-right") == 0)
+    return 7;
   else
     return atoi (name);
 }
